clients/src/main/java/org/apache/kafka/common/config/AbstractConfig.java

```diff
@@ -355,6 +355,29 @@ public void logUnused() {
             log.warn("The configuration '{}' was supplied but isn't a known config.", key);
     }
 
+    private <T> T getConfiguredInstance(Object klass, Class<T> t, Map<String, Object> configPairs) {
+        if (klass == null)
+            return null;
+
+        Object o;
+        if (klass instanceof String) {
```

Comment:
Some of the internal awkwardness seems to be the result of not having a clear "list of classes" type. Not something we have to do here, but potential room for improvement.

Reply:
Agreed. I will look into that in a follow-up ticket.